### PR TITLE
Generate documentation of webSocket automatically

### PR DESCRIPTION
**Pull request by 315157973**
Master Issue: #10041

### Motivation

Generate documentation of webSocket automatically

### CODE REVIEW

**Comment by 315157973** [comment]
@Anonymitaet Please help check the doc in code

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 1]
```diff
@@ -42,94 +42,124 @@
     // Number of threads used by Global ZK
     public static final int GLOBAL_ZK_THREADS = 8;
 
-    // Name of the cluster to which this broker belongs to
-    @FieldContext(required = true)
+    @FieldContext(required = true, doc = "Name of the cluster to which this broker belongs to")
     private String clusterName;
 
-    // Pulsar cluster url to connect to broker (optional if configurationStoreServers present)
+    @FieldContext(doc = "The HTTPS rest service url to connect to broker")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:48]:
```suggestion
    @FieldContext(doc = "The HTTPS REST service URL to connect to broker")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 2]
```diff
@@ -42,94 +42,124 @@
...
     private String serviceUrl;
+
+    @FieldContext(doc = "The HTTPS rest service TLS URL")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:51]: REST

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 3]
```diff
@@ -42,94 +42,124 @@
...
     private String serviceUrlTls;
+
+    @FieldContext(doc = "The broker binary service url (for produce and consume operations)")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:54]: pls update "url" to "URL" for all occurencies

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 4]
```diff
@@ -42,94 +42,124 @@
...
+            doc = "Configuration Store connection string",
+            deprecated = true
+    )
     private String globalZookeeperServers;
+
+    @FieldContext(doc = "Connection string of configuration store servers")
     private String configurationStoreServers;
-    // Zookeeper session timeout in milliseconds
+
+    @FieldContext(doc = "Zookeeper session timeout in milliseconds")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:74]: ZooKeeper 
pls check all occurrences and update all

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 5]
```diff
@@ -42,94 +42,124 @@
...
+
+    @FieldContext(doc = "Port to use to server HTTPS request")
     private Optional<Integer> webServicePortTls = Optional.empty();
-    // Hostname or IP address the service binds on, default is 0.0.0.0.
+
+    @FieldContext(doc = "Hostname or IP address the service binds on, default is 0.0.0.0.")
     private String bindAddress;
-    // The maximum size of a text message during parsing in WebSocket proxy
+
+    @FieldContext(doc = "The maximum size of a text message during parsing in WebSocket proxy")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:89]:
```suggestion
    @FieldContext(doc = "Maximum size of a text message during parsing in WebSocket proxy")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 6]
```diff
@@ -42,94 +42,124 @@
...
+
+    @FieldContext(doc = "Authentication provider name list, which is a list of class names")
     private Set<String> authenticationProviders = Sets.newTreeSet();
-    // Enforce authorization
+
+    @FieldContext(doc = "Enforce authorization")
     private boolean authorizationEnabled;
-    // Authorization provider fully qualified class-name
+
+    @FieldContext(doc = "Authorization provider fully qualified class-name")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:101]:
```suggestion
    @FieldContext(doc = "Authorization provider fully qualified class name")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 7]
```diff
@@ -42,94 +42,124 @@
...
     private String authorizationProvider = PulsarAuthorizationProvider.class.getName();
 
-    // Role names that are treated as "super-user", meaning they will be able to
-    // do all admin operations and publish/consume from all topics
+    @FieldContext(doc = "Role names that are treated as \"super-user\", "
+            + "meaning they will be able to do all admin operations and publish/consume from all topics")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:105]:
```suggestion
    @FieldContext(doc = "Role names that are treated as \"super-user\", "
            + "which means they can do all admin operations and publish to or consume from all topics")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 8]
```diff
@@ -42,94 +42,124 @@
...
     private Set<String> superUserRoles = Sets.newTreeSet();
 
-    // Allow wildcard matching in authorization
-    // (wildcard matching only applicable if wildcard-char:
-    // * presents at first or last position eg: *.pulsar.service, pulsar.service.*)
+    @FieldContext(doc = "Allow wildcard matching in authorization "
+            + "(wildcard matching only applicable if wildcard-char: "
+            + "presents at first or last position eg: *.pulsar.service,pulsar.service.*)")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:110]: do not use `eg` in tech writing. instead, use `for example`

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 9]
```diff
@@ -42,94 +42,124 @@
...
     private boolean authorizationAllowWildcardsMatching = false;
 
-    // Authentication settings of the proxy itself. Used to connect to brokers
+    @FieldContext(doc = "Authentication settings of the proxy itself. Used to connect to brokers")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:113]:
```suggestion
    @FieldContext(doc = "Proxy authentication settings used to connect to brokers")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 10]
```diff
@@ -42,94 +42,124 @@
...
     private String brokerClientAuthenticationPlugin;
+
+    @FieldContext(doc = "Authentication parameters of the proxy itself. Used to connect to brokers")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:116]: same

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 11]
```diff
@@ -42,94 +42,124 @@
...
     private String brokerClientAuthenticationParameters;
-    // Path for the trusted TLS certificate file for outgoing connection to a server (broker)
+
+    @FieldContext(doc = "Path for the trusted TLS certificate file for outgoing connection to a server (broker)")
     private String brokerClientTrustCertsFilePath = "";
 
-    // Number of IO threads in Pulsar Client used in WebSocket proxy
+    @FieldContext(doc = "Number of IO threads in Pulsar Client used in WebSocket proxy")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:122]: client

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 12]
```diff
@@ -42,94 +42,124 @@
...
     private int webSocketNumIoThreads = Runtime.getRuntime().availableProcessors();
 
-    // Number of threads to use in HTTP server
+    @FieldContext(doc = "Number of threads to use in HTTP server")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:125]: used

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 13]
```diff
@@ -42,94 +42,124 @@
...
     private int numHttpServerThreads = Math.max(6, Runtime.getRuntime().availableProcessors());
 
-    // Number of connections per Broker in Pulsar Client used in WebSocket proxy
+    @FieldContext(doc = "Number of connections per Broker in Pulsar Client used in WebSocket proxy")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:128]: broker

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 14]
```diff
@@ -42,94 +42,124 @@
...
     private int webSocketConnectionsPerBroker = Runtime.getRuntime().availableProcessors();
-    // Time in milliseconds that idle WebSocket session times out
+
+    @FieldContext(doc = "Time in milliseconds that idle WebSocket session times out")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:131]:
```suggestion
    @FieldContext(doc = "Timeout of idling WebSocket session (in millisecond)")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 15]
```diff
@@ -42,94 +42,124 @@
...
     private String tlsTrustCertsFilePath = "";
-    // Accept untrusted TLS certificate from client
+
+    @FieldContext(doc = "Accept untrusted TLS certificate from client")
     private boolean tlsAllowInsecureConnection = false;
-    // Specify whether Client certificates are required for TLS
-    // Reject the Connection if the Client Certificate is not trusted.
+
+    @FieldContext(doc = "Specify whether Client certificates are required for "
+            + "TLS Reject the Connection if the Client Certificate is not trusted")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:157]:
```suggestion
    @FieldContext(doc = "Specify whether client certificates are required for "
            + "TLS rejecting the connection if the client certificate is not trusted")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 16]
```diff
@@ -42,94 +42,124 @@
...
     private boolean tlsRequireTrustedClientCertOnConnect = false;
-    // Tls cert refresh duration in seconds (set 0 to check on every new connection) 
+
+    @FieldContext(doc = "Tls cert refresh duration in seconds (set 0 to check on every new connection)")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:160]:
```suggestion
    @FieldContext(doc = "TLS cert refresh duration (in second). 0 means checking every new connection.")
```

File: pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java [hunk 17]
```diff
@@ -42,94 +42,124 @@
...
     private long tlsCertRefreshCheckDurationSec = 300;
 
+    @FieldContext(doc = "key-value properties, types are all String")
```
Comment (Anonymitaet) [inline pulsar-websocket/src/main/java/org/apache/pulsar/websocket/service/WebSocketProxyConfiguration.java:163]:
```suggestion
    @FieldContext(doc = "Key-value properties. Types are all String")
```

**Review by Anonymitaet — CHANGES_REQUESTED** [review]

**Review by Anonymitaet — APPROVED** [review]
Thanks for your contribution. Approved from the language perspective.

**Comment by 315157973** [comment]
@tuteng  PTAL

**Review by sijie — APPROVED** [review]

**Pull request merged**